packages/console-extension/src/index.ts

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import {
+  ILabStatus,
+  IRouter,
+  JupyterFrontEndPlugin,
+  JupyterFrontEnd,
+  Router
+} from '@jupyterlab/application';
+
+import { CommandToolbarButton, IThemeManager, Toolbar } from '@jupyterlab/apputils';
+
+import { IConsoleTracker } from '@jupyterlab/console';
+
+import { ITranslator } from '@jupyterlab/translation';
+
+import { clearIcon, refreshIcon, runIcon } from '@jupyterlab/ui-components';
+
+import { SingleWidgetApp } from '@jupyterlite/application';
+
+import { liteIcon } from '@jupyterlite/ui-components';
+
+import { Panel, Widget } from '@lumino/widgets';
+
+/**
+ * A plugin to add buttons to the console toolbar.
+ */
+const buttons: JupyterFrontEndPlugin<void> = {
+  id: 'replite:buttons',
+  autoStart: true,
+  requires: [ITranslator],
+  optional: [IConsoleTracker],
+  activate: (
+    app: JupyterFrontEnd,
+    translator: ITranslator,
+    tracker: IConsoleTracker | null
+  ) => {
+    if (!tracker) {
+      return;
+    }
+
+    const { commands } = app;
+    const trans = translator.load('jupyterlab');
+
+    // wrapper commands to be able to override the icon
+    const runCommand = 'replite:run';
+    commands.addCommand(runCommand, {
+      caption: trans.__('Run'),
+      icon: runIcon,
+      execute: () => {
+        return commands.execute('console:run-forced');
+      }
+    });
+
+    const runButton = new CommandToolbarButton({
+      commands,
+      id: runCommand
+    });
+
+    const restartCommand = 'replite:restart';
+    commands.addCommand(restartCommand, {
+      caption: trans.__('Restart'),
+      icon: refreshIcon,
+      execute: () => {
+        return commands.execute('console:restart-kernel');
+      }
+    });
+
+    const restartButton = new CommandToolbarButton({
+      commands,
+      id: restartCommand
+    });
+
+    const clearCommand = 'replite:clear';
+    commands.addCommand(clearCommand, {
+      caption: trans.__('Clear'),
+      icon: clearIcon,
+      execute: () => {
+        return commands.execute('console:clear');
+      }
+    });
+
+    const clearButton = new CommandToolbarButton({
+      commands,
+      id: clearCommand
+    });
+
+    tracker.widgetAdded.connect((_, console) => {
+      const { toolbar } = console;
+
+      console.toolbar.addItem('run', runButton);
+      console.toolbar.addItem('restart', restartButton);
+      console.toolbar.addItem('clear', clearButton);
+
+      toolbar.addItem('spacer', Toolbar.createSpacerItem());
+
+      const wrapper = new Panel();
```

Comment:
i :heart: branding, but would consider:
- toning it down to just an icon + tooltip
- making it a separate plugin which can be independently disabled

Reply:
> toning it down to just an icon + tooltip

done in https://github.com/jupyterlite/jupyterlite/pull/498/commits/01a5564622413e4fb2f3ba1ffc26657dc3ab9a2e

Reply:
looks good. i feel like perhaps if `?kernel` is not specified, then the kernel picker should probably still be shown, as the once you pick it with the dialog, there's no way to change it. Perhaps collapsing the kernel text down to just the kernel status icon might be a good way.

Reply:
> if ?kernel is not specified, then the kernel picker should probably still be shown

Yes that's the default behavior. 

Then we could indeed as the toolbar widget to select the kernel. Otherwise it's still possible to open the command palette and choose "Change Kernel" (although harder to do on mobile)

Reply:
> the default behavior.

Right, what i mean is the one that's in the menu bar in classic that says, e.g. `Python 3 ( )`. I think we could collapse it down to just being the kernel icon and the name in a mouseover, next to the status icon...

> open the command palette 

yeah, i'd basically assume nobody is going to find _any_ keyboard shortcut, unless it is labeled in the tooltip of a button, or otherwise discoverable. indeed, having a _show keyboard shortcuts_ button it probably a _really_ good idea as a default...

really, the whole embeddable toolbar needs some of settings-data-driven love, as i could see it enabling a host of nice things, but maybe not all the time for every use case.

as an aside, not to fix right now: a lot of modals/popups are just going to look _bad_ when embedded due to some hard-coded design decisions based on having the whole screen...